lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -614,7 +615,14 @@ export abstract class ClientApplication {
                 const isServerError = e instanceof ServerError;
                 const isInteractionRequiredError = e instanceof InteractionRequiredAuthError;
                 const isInvalidGrantError = (e.errorCode === BrowserConstants.INVALID_GRANT_ERROR);
-                if (isServerError && isInvalidGrantError && !isInteractionRequiredError) {
+
+                /*
+                 * if RT has expired and SilentTokenRetrievalStrategy is not set to use only existing RT,
+                 * then renew the RT with a hidden iFrame (which uses the network)
+                 */
+                if (isServerError && isInvalidGrantError && !isInteractionRequiredError
+                    && (request.silentTokenRetrievalStrategy !== SilentTokenRetrievalStrategy.NetworkWithExistingRefreshTokenOnly)
+                    && (request.silentTokenRetrievalStrategy !== SilentTokenRetrievalStrategy.CacheOrExistingRefreshToken)) {
```

Comment:
Can you add a log statement if either of these retrieval strategies are true, and print a log statement?

For example:

```typescript
if (isServerError && isInvalidGrantError && !isInteractionRequiredError
                    && (request.silentTokenRetrievalStrategy !== SilentTokenRetrievalStrategy.NetworkWithExistingRefreshTokenOnly)
                    && (request.silentTokenRetrievalStrategy !== SilentTokenRetrievalStrategy.CacheOrExistingRefreshToken)) {
    // ...
}

if (request.silentTokenRetrievalStrategy === SilentTokenRetrievalStrategy.NetworkWithExistingRefreshTokenOnly || request.silentTokenRetrievalStrategy === SilentTokenRetrievalStrategy.CacheOrExistingRefreshToken) {
    this.logger.info("SilentTokenRetrievalStrategy set to not renew expired refresh token.");
}
````